yt/yt/server/node/exec_node/job_environment.cpp

```diff
@@ -1095,6 +1096,17 @@ class TCriJobEnvironment
         }
         spec->Credentials.Gid = ::getgid();
 
+        if (!spec->Environment.contains("USER")) {
+            TString username;
```

Comment:
I'm not sure if I got it correctly. Why isn't it sufficient just to add environment variables in TCriUserJobEnvironment::TCriUserJobEnvironment() like you did and we have to do it twice?

Reply:
Here we fill environemtn for job-proxy.
TCriUserJobEnvironment fills environement for user-job.

Reply:
Oh, I see.